### PR TITLE
Tidy changelog

### PR DESCRIPTION
**Pull request by trbromley**
This PR tidies the changelog and some docstrings of newly-added features.

Changes to the changelog include:
- Move some additions to "Improvements"
- Group new features into sections
- Rearrange the new features sections to match the order we want to advertise them (e.g., pulses first).

Things that we also might want to do:
- Tidy the `<h4>` level subsections in the features section, e.g., update wording, make sure examples are on point.
- Order the improvements according to what we think is most impactful

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -4,36 +4,63 @@
...
+  YY = qml.PauliY(0) @ qml.PauliY(0)
+  ZZ = qml.PauliZ(0) @ qml.PauliZ(1)
+
+  H =  2 * XX + f1 * YY + f2 * ZZ
   ```
+  ```pycon
+  >>> H
+  ParametrizedHamiltonian: terms=3
+  >>> params = [1.2, [2.3, 3.4]]
+  >>> H(params, t=0.5)
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:30]:
```suggestion
asdasdasdasd
```

File: doc/releases/changelog-dev.md [hunk 2]
```diff
@@ -4,39 +4,70 @@
...
-  1: ─╰RY(0.20)─╰SWAP─╭●────────╭SWAP─╰RY(0.09)─╰SWAP─╭●────────╭SWAP─╰RY(0.62)─╰SWAP─┤  State
-  2: ─╭●────────╭SWAP─╰RY(0.68)─╰SWAP─╭●────────╭SWAP─╰RY(0.34)─╰SWAP─╭●────────╭SWAP─┤  State
-  3: ─╰RY(0.92)─╰SWAP─╭●────────╭SWAP─╰RY(0.82)─╰SWAP─╭●────────╭SWAP─╰RY(0.52)─╰SWAP─┤  State
-  4: ─────────────────╰RY(0.81)─╰SWAP─────────────────╰RY(0.06)─╰SWAP─────────────────┤  State
+  >>> coeffs = [2, f1, f2]
+  >>> ops = [XX, YY, ZZ]
+  >>> H =  qml.ops.dot(coeffs, ops)
+  ```
+
+* A `ParametrizedHamiltonian` can be time-evolved by using `ParametrizedEvolution`.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:44]: Worth putting a code example maybe?

File: doc/releases/changelog-dev.md [hunk 3]
```diff
@@ -183,134 +182,149 @@
...
+  >>> print(weights)
+  tensor([0.20308242, 0.91906199, 0.67988804, 0.81290256, 0.08708985,
+          0.81860084, 0.34448344, 0.05655892, 0.61781612, 0.51829044], requires_grad=True)
+  >>> qml.draw(swap_network_circuit, expansion_strategy = 'device')()
+  0: ─╭●────────╭SWAP─────────────────╭●────────╭SWAP─────────────────╭●────────╭SWAP─┤  State
+  1: ─╰RY(0.20)─╰SWAP─╭●────────╭SWAP─╰RY(0.09)─╰SWAP─╭●────────╭SWAP─╰RY(0.62)─╰SWAP─┤  State
+  2: ─╭●────────╭SWAP─╰RY(0.68)─╰SWAP─╭●────────╭SWAP─╰RY(0.34)─╰SWAP─╭●────────╭SWAP─┤  State
+  3: ─╰RY(0.92)─╰SWAP─╭●────────╭SWAP─╰RY(0.82)─╰SWAP─╭●────────╭SWAP─╰RY(0.52)─╰SWAP─┤  State
+  4: ─────────────────╰RY(0.81)─╰SWAP─────────────────╰RY(0.06)─╰SWAP─────────────────┤  State
   ```
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:218]: lol SWAP SWAP SWAP SWAP SWAP SWAP SWAP SWAP SWAP SWAP SWAP SWAP

File: doc/releases/changelog-dev.md [hunk 4]
```diff
@@ -183,134 +182,149 @@
...
+* `qml.math.matmul` now supports PyTorch and Autograd tensors.
+  [(#3613)](https://github.com/PennyLaneAI/pennylane/pull/3613)
+
+* `qml.math.size` now supports PyTorch tensors.
+  [(#3606)](https://github.com/PennyLaneAI/pennylane/pull/3606)
+
+* `qml.purity` can now be used as a measurement process.
+  [(#3551)](https://github.com/PennyLaneAI/pennylane/pull/3551)
+
+* Most quantum channels are now fully differentiable on all interfaces.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:231]: Worth being more specific here? Can we give a list of unsupported quantum channels?

File: doc/releases/changelog-dev.md [hunk 5]
```diff
@@ -319,14 +333,14 @@
...
-* Handles breaking networkx version change by selectively skipping a qcut tensorflow-jit test.
+* Handles breaking the `networkx` version change by selectively skipping a `qcut` TensorFlow-JIT test.
   [(#3609)](https://github.com/PennyLaneAI/pennylane/pull/3609)
   [(#3619)](https://github.com/PennyLaneAI/pennylane/pull/3619)
 
-* Fixed the wires for the Y decomposition in the ZX calculus transform.
+* Fixed the wires for the `Y` decomposition in the ZX calculus transform.
   [(#3598)](https://github.com/PennyLaneAI/pennylane/pull/3598)
 
 * `qml.pauli.PauliWord` is now pickle-able.
```
Comment (isaacdevlugt) [inline doc/releases/changelog-dev.md:346]: Improvement or new feature?

**Review by isaacdevlugt — APPROVED** [review]
Made a lot of changed myself, but looks good to me now! Didn't test code examples, but overall the changelog looks good.

**Review by trbromley — COMMENTED** [review]
Thanks @isaacdevlugt! I'm going to merge now since it's a pain to fix the merge conflicts with the rapidly-changing changelog, but I will revisit two of the comments you left next time we do a tidy :+1:

**Pull request merged**